patches/tModLoader/Terraria/ModLoader/ProjectileLoader.cs

```diff
@@ -268,11 +269,13 @@ public static void ReceiveExtraAI(Projectile projectile, byte[] extraAI)
 		catch (IOException e) {
 			string message = $"Error in ReceiveExtraAI for Projectile {projectile.ModProjectile?.FullName ?? projectile.Name}";
 			if (anyGlobals) {
-				message += ", may be caused by one of these GlobalNPCs:";
+				message += ", may be caused by one of these GlobalProjectiles:";
```

Comment:
We can potentially make this message 50% better if we store the last `GlobalProjectile` which had its hook called, and break the loop below after encountering it.

Reply:
That would add overhead to the normal code path, wouldn't it? This is an exception, I feel like this code looping over all is better than making a local variable always run.

Reply:
Oh wait, I see what you mean, since we are culling out GlobalProjectiles that couldn't be the issue.